### PR TITLE
BL-7882 log4j2 with custom logger

### PR DESCRIPTION
**Pull request by zakrzewskam**

### CODE REVIEW

File: api/src/main/java/uk/gov/dvsa/mot/app/LambdaHandler.java [hunk 1]
```diff
@@ -21,7 +23,8 @@
 
     @Override
     public AwsProxyResponse handleRequest(AwsProxyRequest request, Context context) {
-        context.getLogger().log("Entering lambda handler : " + context.getAwsRequestId());
+        CloseableThreadContext.put(JsonLayout.AwsRequestId, context.getAwsRequestId());
```
Comment (ryszardm-kainos) [inline api/src/main/java/uk/gov/dvsa/mot/app/LambdaHandler.java]: We might need to set AwsRequestId in other handlers that are not proxied through here. For example `TradeServiceRequestHandler`

File: api/src/main/java/uk/gov/dvsa/mot/app/logging/CompletableFutureWrapper.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package uk.gov.dvsa.mot.app.logging;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Supplier;
+
+public class CompletableFutureWrapper {
```
Comment (przemos) [inline api/src/main/java/uk/gov/dvsa/mot/app/logging/CompletableFutureWrapper.java]: Please add the justification what it is for in the comments

File: api/src/main/java/uk/gov/dvsa/mot/app/logging/JsonLayout.java [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+import java.nio.charset.Charset;
+import java.sql.Timestamp;
+import java.text.SimpleDateFormat;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+
+@Plugin(name = "JSONLayout", category = "Core", elementType = "layout", printObject = true)
+public class JsonLayout extends AbstractStringLayout {
```
Comment (przemos) [inline api/src/main/java/uk/gov/dvsa/mot/app/logging/JsonLayout.java]: Please add the justification what it is for in the comments. What to expect out of it.

File: api/src/main/java/uk/gov/dvsa/mot/app/logging/ThreadContextSupplier.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package uk.gov.dvsa.mot.app.logging;
+
+import org.apache.logging.log4j.ThreadContext;
+
+import java.util.Map;
+import java.util.function.Supplier;
+
+public class ThreadContextSupplier<T> implements Supplier<T> {
```
Comment (przemos) [inline api/src/main/java/uk/gov/dvsa/mot/app/logging/ThreadContextSupplier.java]: Please add justification what it is for and why in the comments

File: api/src/main/java/uk/gov/dvsa/mot/app/logging/JsonLayout.java [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+    public static final String AwsRequestIdKey = "AWSRequestId";
```
Comment (przemos) [inline api/src/main/java/uk/gov/dvsa/mot/app/logging/JsonLayout.java]: do we need to couple JSONLayout with the specific attribute names ? It should remain agnostic as to what content it handles.

File: api/src/main/java/uk/gov/dvsa/mot/app/logging/LoggerParamsManager.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.ThreadContext;
+
+import java.util.Map;
+
+public class LoggerParamsManager {
```
Comment (przemos) [inline api/src/main/java/uk/gov/dvsa/mot/app/logging/LoggerParamsManager.java:12]: Please add justification what it is for and why in the comment

**Review by przemos — APPROVED** [review]

**Pull request merged**